include/graphblas/bsp1d/blas1.hpp

```diff
@@ -1770,10 +1808,13 @@ namespace grb {
 
 		// catch execute
 		if( phase != RESIZE ) {
-			assert( phase == EXECUTE );
-			if( ret == FAILED ) {
```

Comment:
This here is the most common pattern though. The code block pertains to either EXECUTE or TRY, so FAILED is a possible valid return code (from the TRY). In those cases, the global number of nonzeroes needs to be synced while the FAILED error code is still returned. This is not necessary if the underlying implementation guarantees that vector outputs always have the same full capacity -- in that case, FAILED would never be returned.

This pattern repeats many times in the code that follows, sometimes with slight variations